rad/prelude/validation.rad

```diff
@@ -79,3 +79,24 @@
   (fn [v]
     (fn [xs]
       (map v xs))))
+
+(def validator/uuid
+  "Validates UUIDs."
+  (validator/and
+   [(validator/type :string)
+    (validator/pred "valid UUID" uuid?)]))
+
+(def validator/signed
+  "Checks that a value is a dict with `:signature` and `:author` keys, and that
+  the signature is valid for the rest of the dict for that author. The rest of
+  the dict is turned into a string according to `show`."
+  (validator/and
+   [(validator/keys
+     {:author (validator/pred "valid public key" public-key?)})
+    (validator/pred
+     "Valid signature"
+     (fn [d]
+       (verify-signature
+        (lookup :author d)
+        (lookup :signature d)
+        (show (delete :author (delete :signature d))))))]))
```

Comment:
Test please

Reply:
I added a test but it doesn't work. I suspect the `catch` has something to do with it.